activerecord/test/cases/connection_adapters/connection_handler_test.rb

```diff
@@ -253,6 +310,40 @@ def test_forked_child_doesnt_mangle_parent_connection
           assert_equal 3, ActiveRecord::Base.connection.select_value("SELECT COUNT(*) FROM people")
         end
 
+        def test_reaper_does_not_reap_parent_connection_pool_after_fork
+          frequency = 0.001
+          @handler.establish_connection(@config.merge(reaping_frequency: frequency))
+
+          assert_equal 1, ActiveRecord::ConnectionAdapters::ConnectionPool::Reaper.pools[frequency].size
```

Comment:
(similar to suggestion above)

I feel like this is testing the implementation rather than behaviour. I don't _think_ there's a behaviour changes from this PR (please correct me if I'm wrong 😅), we're just avoiding some cycles (and objects?) by cleaning up better, so I don't think we need a test.

Reply:
yup this PR ensures that we avoid looping through discarded connection pools unnecessary in the reaper so there isn’t a behaviour change. However, this might be a problem in multi-tenant Rails apps where hundreds of connection pools are used so I think we should clean up properly in the reaper. In Rails 5.2, a reaper isn’t started for discarded pools but it was using one thread per pool which was a much bigger problem which you’ve since fixed in 6.0 :)

Reply:
@jhawthorn I re-read this and was wondering why you think testing implementation in this case might not be worth it. I'm not sure if we can actually test this at the acceptance test level since this involves a class instance variable that is copied due to forking.

Reply:
Actually I think I would fall back to my approach of de-registering the pool from the reaper during `discard!` which happens after a fork. I've confirmed that `discard!` always happen before the reaper is started after a fork. De-registering it ensures that we don't spend any iterations on discarded pools in the reaper which is much more similar to the behavior in 5.2.